pkg/plugin/processor/builtin/webhook_http_test.go

```diff
@@ -0,0 +1,483 @@
+			wantErr: "missing required parameter 'url'",
+		},
+		{
+			name: "empty url returns error",
+			config: map[string]string{
+				"request.url": "",
+			},
+			wantErr: "missing required parameter 'url'",
+		},
+		{
+			name: "invalid url returns error",
+			config: map[string]string{
+				"request.url": ":not/a/valid/url",
+			},
+			wantErr: "configuration check failed: parse \":not/a/valid/url\": missing protocol scheme",
+		},
+		{
+			name: "invalid method returns error",
+			config: map[string]string{
+				"request.url":    "http://example.com",
+				"request.method": ":foo",
+			},
+			wantErr: "configuration check failed: net/http: invalid method \":foo\"",
+		},
+		{
+			name: "invalid backoffRetry.count returns error",
+			config: map[string]string{
+				"request.url":        "http://example.com",
+				"backoffRetry.count": "not-a-number",
+			},
+			wantErr: `failed parsing configuration: failed decoding map: 1 error(s) decoding:
+
+* cannot parse 'backoffRetry.count' as float: strconv.ParseFloat: parsing "not-a-number": invalid syntax`,
+		},
+		{
+			name: "invalid backoffRetry.min returns error",
+			config: map[string]string{
+				"request.url":        "http://example.com",
+				"backoffRetry.count": "1",
+				"backoffRetry.min":   "not-a-duration",
+			},
+			wantErr: `failed parsing configuration: failed decoding map: 1 error(s) decoding:
+
+* error decoding 'backoffRetry.min': time: invalid duration "not-a-duration"`,
+		},
+		{
+			name: "invalid backoffRetry.max returns error",
+			config: map[string]string{
+				"request.url":        "http://example.com",
+				"backoffRetry.count": "1",
+				"backoffRetry.max":   "not-a-duration",
+			},
+			wantErr: `failed parsing configuration: failed decoding map: 1 error(s) decoding:
+
+* error decoding 'backoffRetry.max': time: invalid duration "not-a-duration"`,
+		},
+		{
+			name: "invalid backoffRetry.factor returns error",
+			config: map[string]string{
+				"request.url":         "http://example.com",
+				"backoffRetry.count":  "1",
+				"backoffRetry.factor": "not-a-number",
+			},
+			wantErr: `failed parsing configuration: failed decoding map: 1 error(s) decoding:
+
+* cannot parse 'backoffRetry.factor' as float: strconv.ParseFloat: parsing "not-a-number": invalid syntax`,
+		},
+		{
+			name: "valid url returns processor",
+			config: map[string]string{
+				"request.url": "http://example.com",
+			},
+			wantErr: "",
+		},
+		{
+			name: "valid url and method returns processor",
+			config: map[string]string{
+				"request.url":    "http://example.com",
+				"request.method": "GET",
+			},
+			wantErr: "",
+		},
+		// {
+		//	name: "invalid backoff retry config is ignored",
+		//	config: map[string]string{
+		//		"request.url":                 "http://example.com",
+		//		"backoffRetry.min":    "not-a-duration",
+		//		"backoffRetry.max":    "not-a-duration",
+		//		"backoffRetry.factor": "not-a-number",
+		//	},
+		//	wantErr: "something",
+		// },
```

Comment:
I copied this over from the legacy httprequest processor, so that we can discuss it here. I wasn't sure what's the reason for ignoring errors on optional parameters. IMO, if a value is provided, then it must be valid, because there was a clear intention to use it. If we silently set it to a default value, it may lead to unexpected behavior.

Reply:
We discussed this on a standup, all provided options (required or non-required) need to be validated.